app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -370,15 +329,56 @@ class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsTypeItem.C
         fun bindItem(contribution: Contribution) {
             view.setTitle(contribution.description)
             view.setDescription(contribution.title)
-            view.setImageUrl(contribution.imageUrl)
             view.setIcon(contribution.editType)
-            view.setPageViewCountText(contribution.pageViews)
             getPageView(view, contribution)
+            getImageDetails(view, contribution)
         }
 
-        private fun getPageView(view: SuggestedEditsContributionsItemView, contribution: Contribution) {
+        private fun getImageDetails(itemView: SuggestedEditsContributionsItemView, contribution: Contribution) {
             val disposables = CompositeDisposable()
+            if (contribution.editType == EDIT_TYPE_ARTICLE_DESCRIPTION) {
+                disposables.add(ServiceFactory.getRest(contribution.wikiSite).getSummary(null, contribution.title)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ summary ->
+                            contribution.imageUrl = summary.thumbnailUrl.toString()
+                            itemView.setImageUrl(contribution.imageUrl)
+                        }, { t ->
+                            L.e(t)
+                        }))
+            } else {
+                disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getImageInfo(contribution.title, "en")
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ response ->
+                            val page = response.query()!!.pages()!![0]
+                            val labels = ArrayList<String>()
+                            if (page.imageInfo() != null) {
+                                val imageInfo = page.imageInfo()!!
+                                contribution.description = imageInfo.metadata!!.imageDescription()
+                                contribution.imageUrl = imageInfo.originalUrl
+                                if (contribution.editType == EDIT_TYPE_IMAGE_TAG) {
+                                    if (!page.imageLabels.isNullOrEmpty()) {
+                                        for (imageLabel in page.imageLabels) {
+                                            labels.add(imageLabel.label)
+                                        }
+                                        contribution.description = labels.joinToString(" , ")
```

Comment:
It can be simplified to `page.imageLabels.joinToString(" , ")`

Reply:
@cooltey we need page.imageLabels.label

Reply:
oops. you're right